dokka-runners/dokka-gradle-plugin/src/main/kotlin/internal/PluginFeaturesService.kt

```diff
@@ -264,3 +306,58 @@ internal abstract class PluginFeaturesService : BuildService<PluginFeaturesServi
         }
     }
 }
+
+
+/**
+ * Determine if Gradle is generating DSL script accessors for precompiled script plugins.
+ *
+ * When Gradle generates accessors, it creates an empty project in a temporary directory and runs no tasks.
+ * So, we can guess whether Gradle is generating accessors based on this information.
+ */
+private fun Project.isGradleGeneratingAccessors(): Boolean {
+    if (gradle.rootProject.name != "gradle-kotlin-dsl-accessors") {
+        return false
+    }
+
+    if (gradle.taskGraph.allTasks.isNotEmpty()) {
+        return false
+    }
+
+    /**
+     * When a Gradle build is executed with no requested tasks and no arguments then
+     * Gradle runs a single 'default' task that has no values.
+     */
+    fun TaskExecutionRequest.isDefaultTask(): Boolean =
+        projectPath == null && args.isEmpty() && rootDir == null
+
+    val taskRequest = gradle.startParameter.taskRequests.singleOrNull() ?: return false
+    if (!taskRequest.isDefaultTask()) return false
+
+    val rootProjectPath = gradle.rootProject.rootDir.invariantSeparatorsPath
+    return rootProjectPath
+        .substringBeforeLast("/")
+        .endsWith("build/tmp/generatePrecompiledScriptPluginAccessors")
+}
+
+
+/**
+ * Walk up the file tree until we discover a `gradle.properties` file.
+ */
+private fun Project.findGradlePropertiesFile(): File? {
```

Comment:
Oh, this is very hacky... not sure what I think about it

The main question from me: is there any possibility of false-positives? looks like there are enough guards, but still want to have some thoughts on it

Reply:
Thanks! sounds reasonable

> Release multiple versions of DGP, with separate suffixes
> Or even with (slightly) more descriptive suffixes, like 2.0.0-v2-migration and 2.0.0-v2.

let's try not to do this. I believe current approach should be enough for migration period

BTW, until this is merged, the workaround is to apply plugin like `plugins.apply(DokkaHtmlPlugin::class)` and configure extension via `extensions.configure<DokkaExtension>("dokka") {}` ([example](https://github.com/whyoleg/cryptography-kotlin/blob/f2640426484e334faa7f5db2e6ae1abc691fd367/build-logic/src/main/kotlin/ckbuild.documentation.gradle.kts)) - in this case no warning is emitted